tests/common/widgets/search-box/options.ts

```diff
@@ -307,6 +307,74 @@ export function createOptionsTests(
       expect(screen.getByRole('searchbox')).toHaveValue('iPhone');
     });
 
+    test('refines query only when composition is complete', async () => {
+      const searchClient = createSearchClient({});
+
+      await setup({
+        instantSearchOptions: {
+          indexName: 'indexName',
+          searchClient,
+        },
+        widgetParams: {},
+      });
+
+      // Typing 木 using the Wubihua input method
+      // see:
+      // - https://en.wikipedia.org/wiki/Stroke_count_method
+      // - https://developer.mozilla.org/en-US/docs/Web/API/Element/compositionend_event
+      const character = '木';
+      const strokes = ['一', '丨', '丿', '丶', character];
+
+      await act(async () => {
+        await wait(0);
+
+        searchClient.search.mockClear();
```

Comment:
```suggestion
        castToJestMock(searchClient.search).mockClear();
```

also why not right after setup + wait 0, before starting to compose?

```
await act(async () => {
  await wait(0);
  castToJestMock(searchClient.search).mockClear();
});

// ... character etc
```

Reply:
I need to wrap the events inside of `act()` so I'm reusing the one that waits for initialization.